packages/office-ui-fabric-react/src/utilities/color/cssColor.ts

```diff
@@ -1,42 +1,67 @@
 import { IRGB } from './interfaces';
 import { MAX_COLOR_ALPHA } from './consts';
-import { COLOR_VALUES } from './colorValues';
 import { hsl2rgb } from './hsl2rgb';
 
 /**
  * Converts a valid CSS color string to an RGB color.
  * Note that hex colors *must* be prefixed with # to be considered valid.
  * Alpha in returned color defaults to 100.
+ * Four and eight digit hex values (with alpha) are supported if the current browser supports them.
  */
 export function cssColor(color: string): IRGB | undefined {
-  if (!color) {
-    return undefined;
+  // Need to check the following valid color formats: RGB(A), HSL(A), hex, named color
+
+  // First check for RGB(A) and HSL(A) formats
+  const rgbaColor = _rgba(color);
+  if (rgbaColor) {
+    return rgbaColor;
+  }
+  const hslaColor = _hsla(color);
+  if (hslaColor) {
+    return hslaColor;
   }
-  return _named(color) || _hex3(color) || _hex6(color) || _rgba(color) || _hsla(color);
-}
 
-/**
- * If `str` is a valid HTML color name, returns an RGB color (with alpha 100).
- * Otherwise returns undefined.
- */
-function _named(str: string): IRGB | undefined {
-  const c = (COLOR_VALUES as any)[str.toLowerCase()];
+  // Otherwise, do a catch-all for hex and named colors
+  // This element must be attached to the DOM for getComputedStyle() to have a value
+  const elem = document.createElement('div');
```

Comment:
Can you move all this into a `_named` function, then restore the overall `cssColor` method contents to what it was before? 

Also, to get around the ssr-tests error, `_named` should check at the beginning whether `document` is defined, and return undefined if not.

Reply:
this isn't just for named colors though, it works for everything, and will handle hex colors too and things like rgb(0,0,0,1) which is technically invalid but the browser handles it fine
checking rgba and hsla at the top are just for convenience, to make the later error-checking logic easier, and presumably for perf so that you don't always have to create and append and read an element (which may or may not cause a redraw...)

i think returning undefined when there's no `document` will still fail the browser test because then swatch color picker won't be able to render! but i can try it. the additional question is, do we care at this point?

Reply:
Do you know who owns the ssr-tests?

Reply:
I'd still like to have the getComputedStyle logic in its own function (by whatever name makes sense), just for readability/cleanliness/whatever. I also don't think it hurts to keep the old hex parsing logic as a fallback for the server-side rendering case (it's not very large).

You'd have to look at the git history to see who added ssr-tests.

Reply:
fair enough, brought back the hex stuff, which conveniently fixes the ssr-tests because those probably use hex as well, so it never reaches the document.* code